### PR TITLE
Modify code comments in write.h

### PR DESCRIPTION
**Pull request by dota17**
Add missing file write.h in issue #985

### CODE REVIEW

File: include/json/writer.h [hunk 1]
```diff
@@ -28,29 +28,29 @@ class Value;
 
 /**
```
Comment (BillyDonahue) [inline include/json/writer.h]: needs star here to connect to the /** above.

File: include/json/writer.h [hunk 2]
```diff
@@ -28,29 +28,29 @@ class Value;
...
-      factory.newStreamWriter());
-    writer->write(value, &std::cout);
-    std::cout << std::endl;  // add lf and flush
-  }
-\endcode
+* Usage:
+*  \code
+*  using namespace Json;
+*  void writeToStdout(StreamWriter::Factory const& factory, Value const& value) {
+*    std::unique_ptr<Json::StreamWriter> const writer(
```
Comment (BillyDonahue) [inline include/json/writer.h]: "using namespace Json" above makes this unnecessary.

File: include/json/writer.h [hunk 3]
```diff
@@ -28,29 +28,29 @@ class Value;
...
-      Do not take ownership of sout, but maintain a reference during function.
-      \pre sout != NULL
-      \return zero on success (For now, we always return zero, so check the
-     stream instead.) \throw std::exception possibly, depending on configuration
-   */
+/**    Write Value into document as configured in sub-class.
+*      Do not take ownership of sout, but maintain a reference during function.
+*      \pre sout != NULL
+*      \return zero on success (For now, we always return zero, so check the
+*     stream instead.) \throw std::exception possibly, depending on configuration
```
Comment (BillyDonahue) [inline include/json/writer.h]: indented too much.

File: include/json/writer.h [hunk 4]
```diff
@@ -73,49 +73,49 @@ String JSON_API writeString(StreamWriter::Factory const& factory,
...
-  StreamWriterBuilder builder;
-  builder["commentStyle"] = "None";
-  builder["indentation"] = "   ";  // or whatever you like
-  std::unique_ptr<Json::StreamWriter> writer(
-      builder.newStreamWriter());
-  writer->write(value, &std::cout);
-  std::cout << std::endl;  // add lf and flush
-\endcode
+* Usage:
+*  \code
```
Comment (BillyDonahue) [inline include/json/writer.h]: indent the \code to \endcode 1 more space to be consistent with reader.h

File: include/json/writer.h (hunk 2 above)
Comment (dota17) [reply include/json/writer.h]: > "using namespace Json" above makes this unnecessary.

Oops, I didn't see it.

File: include/json/writer.h (hunk 4 above)
Comment (dota17) [reply include/json/writer.h]: you mean making \code and \encode are one more space ?

Comment (BillyDonahue) [reply include/json/writer.h]: I mean yes. The other usage or example clauses have 2-space indents for the \code and following associated block, and here there's only a 1-space indent.

File: include/json/writer.h [hunk 5]
```diff
@@ -74,7 +74,7 @@ String JSON_API writeString(StreamWriter::Factory const& factory,
 /** \brief Build a StreamWriter implementation.
 
 * Usage:
-*  \code
+*   \code
 *  using namespace Json;
```
Comment (BillyDonahue) [inline include/json/writer.h]: the code should be indented to be directly below the \code.

**Comment by baylesj** [comment]
Why are we bothering with C-style comments? Why not just move everything over to C++ style comments (e.g. //)?

**Comment by dota17** [comment]
> Why are we bothering with C-style comments? Why not just move everything over to C++ style comments (e.g. //)?

Hi @baylesj  I follow the C style comment in the other place of this project(e.g. read.h value.h), just to keep it consistent. but if you perfer to C++ style, I'd like to apply it  in this whole project, and before that, we should close this PR or merge it.

**Pull request merged**